YourBCABus/Views/BusesView.swift

```diff
@@ -102,7 +102,8 @@ struct BusesView: View {
                     ProgressView("Loading").padding(.vertical, 64)
                 case .some(.success(let result)):
                     let predicate = searchText.isEmpty ? nil : busPredicate(for: searchText)
-                    if let school = result.data?.school, let buses = school.buses.filter { predicate?.evaluate(with: $0) ?? true }.sorted(by: { a, b in
+                    let school = result.data?.school;
+                    if let buses = school?.buses.filter({ predicate?.evaluate(with: $0) ?? true }).sorted(by: { a, b in
```

Comment:
Is there a reason why we can't do:

```suggestion
                    if let school = result.data?.school, let buses = school?.buses.filter({ predicate?.evaluate(with: $0) ?? true }).sorted(by: { a, b in
```

Seems like it would save us from using a bunch of optional chaining operators down the line.

Reply:
No, I just did that because it was what worked for me for getting it to build